### PR TITLE
Add MedGen Prefixes

### PR DESCRIPTION
**Pull request by joeflack4**
## Updates
- Add: MedGen prefixes to prefixes.csv
- Add: config/medgen.yml

## Sub-tasks 
- [x] Add: MedGen prefixes to `prefixes.csv`
- [x] Add: `config/medgen.yml`
- [ ] Add: MedGen prefixes to `mondo.sssom.config.yml` `curie_map` and `extended_prefix_map`
- [ ] Add anywhere else that may be missing as outlined in `docs/developer/add-new-source.md`
- [ ] Make sure nothing was added that shouldn't be, in compliance with #378

## Related
- #375
- #377 
- #146 
- https://github.com/monarch-initiative/mondo-ingest/pull/363#discussion_r1396540100

## Addresses
- #414

### CODE REVIEW

File: src/ontology/config/prefixes.csv [hunk 1]
```diff
@@ -228,4 +228,6 @@ ICD10WHO,https://icd.who.int/browse10/2019/en#/
 OMIMPS,https://www.omim.org/phenotypicSeries/PS
 OMIM,https://omim.org/entry/
 Orphanet,http://www.orpha.net/ORDO/Orphanet_
-GARD,http://purl.obolibrary.org/obo/GARD_
\ No newline at end of file
+MEDGEN,http://purl.obolibrary.org/obo/MEDGEN_
```
Comment (joeflack4) [inline src/ontology/config/prefixes.csv:231]: ## Update: `prefixes.csv` with prefix maps

File: src/ontology/metadata/medgen.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+id: MEDGEN
```
Comment (joeflack4) [inline src/ontology/metadata/medgen.yml:1]: ## Add: `metadata/medgen.yml`
I've marked this PR as 'draft' mainly because this file hasn't been generated correctly. I don't remember at the moment, but I think there is a `make` goal go dynamically generate these `metadata/*.yml` files.

At the time I created this, it was bundled with some other related changes (#375 #363) but now I've made it separate and I also no longer see an impending need for this file quite yet, until MedGen is properly integrated into Mondo.

**Review by joeflack4 — COMMENTED** [review]
At the time I created these changes, it was bundled with some other related changes (#375 #363) but now I've made it separate and I also no longer see an impending need for this file quite yet, until MedGen is properly integrated into Mondo.

So I don't mind if we don't merge this PR or hold off on it until we properly integrate MedGen.

Or we could perhaps just discard the PR, as it will be made obsolete by:
- #146

File: src/ontology/config/prefixes.csv (hunk 1 above)
Comment (matentzn) [reply src/ontology/config/prefixes.csv:231]: Why remove the GARD prefix?

File: src/ontology/metadata/medgen.yml (hunk 2 above)
Comment (matentzn) [reply src/ontology/metadata/medgen.yml:1]: Unless medgen is formally folded in as a source (i.e. mentioned in mondo-ingest-odk.yaml) we should have to add stuff about it?

File: src/ontology/config/prefixes.csv (hunk 1 above)
Comment (joeflack4) [reply src/ontology/config/prefixes.csv:232]: Good to have your eye on this. Looks like I made a copy/paste error. Fixed!

File: src/ontology/metadata/medgen.yml (hunk 2 above)
Comment (joeflack4) [reply src/ontology/metadata/medgen.yml]: I don't remember my exact line of thinking when I made these changes a few weeks ago, but I think it was (i) should make `prefixes.csv` consistent with `mondo.sssom.config.yml`, which has a MedGen prefix. But (ii) I expected MedGen terms to start coming into the pipeline for some reason, and if that's not the case, we don't need a `.yml` for this.

TLDR: I've just deleted the `.yml`

File: src/ontology/metadata/mondo.sssom.config.yml [hunk 3]
```diff
@@ -28,6 +28,8 @@ curie_map:
   ICD10WHO__2: http://apps.who.int/classifications/icd10/browse/2010/en#/
   OMIMPS: https://omim.org/phenotypicSeries/PS
   MEDGEN: http://identifiers.org/medgen/
+  MEDGEN2: http://identifiers.org/medgen/
```
Comment (joeflack4) [inline src/ontology/metadata/mondo.sssom.config.yml:31]: ## Update to `mondo.sssom.config.yml`
I noticed that `mondo.sssom.config` was now not as up-to-date as `prefixes.csv`, so I updated it as well.

<details><summary>More info</summary>
<p>

When doing this, I noticed that actually there are more conflicts between these two files.. ugh, but I'm a bit fatigued from this now and I think maybe I'll leave it be and just hope to get to this one sooner:
- #146

</p>
</details>

**Comment by joeflack4** [comment]
I suppose this is no longer a draft.

File: src/ontology/metadata/mondo.sssom.config.yml (hunk 3 above)
Comment (joeflack4) [reply src/ontology/metadata/mondo.sssom.config.yml]: Just undid these changes as [I've just learned](https://github.com/monarch-initiative/mondo-ingest/pull/375#pullrequestreview-1741751315) that `mondo.sssom.config.yml` comes from `mondo`.
